app/Room.php

```diff
@@ -42,8 +42,22 @@ public function createRoom(string $title, int $roomUrl, string $password, int $a
         $this->save();
     }
 
+    /**
+     * @param int $roomUrl
+     * @return Model|\Illuminate\Database\Query\Builder|object|null
+     */
+
     public function getRoom(int $roomUrl)
     {
         return DB::table('rooms')->where('room_url', '=', $roomUrl)->first();
     }
+
+    /**
+     * @param int $adminUserId
+     * @return Model|\Illuminate\Database\Query\Builder|object|null
+     */
+
+    public function getAdminId () {
+        return DB::table('rooms')->where('admin_user_id', '=', Auth::id())->first();
```

Comment:
This is not okay. This only checks if the user is an admin of any room. You have to add a check that the user is an admin of the given room.